hypershift-operator/controllers/sharedingress/router.go

```diff
@@ -64,7 +64,31 @@ func ReconcileRouterDeployment(deployment *appsv1.Deployment, hypershiftOperator
 				},
 				ServiceAccountName:           RouterServiceAccount().Name,
 				AutomountServiceAccountToken: ptr.To(true),
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "infra",
+						Value:    "true",
+						Effect:   corev1.TaintEffectNoSchedule,
+						Operator: corev1.TolerationOpEqual,
+					},
+				},
 				Affinity: &corev1.Affinity{
+					NodeAffinity: &corev1.NodeAffinity{
+						PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
```

Comment:
Can it be `required` policy instead of `preferred` so we can enforce these to be on different nodes and zones. (ideally we don't expect them to be colocated)